### DISCUSSION TITLE
Join without selecting

### DISCUSSION THREAD

**Discussion by lazaroofarrill** [PostgREST/postgrest#2640] [Q&A] [answered]
Sometimes I need to filter a top level entity based on a relation but I need to return all rows from the relation.
In this case I usually join twice, once with inner to filter and once with left to select from it.

I tried doing this using the rest api but since joins are defined in the where query i have to select the join table even though I'm not using it. Is it possible to do this join without selecting it?

This is how my query looks like

```
GET rest/v1/business_view?
    select=*,categories:business_categories(id, name),
    menus_f:menus!inner(id),
    menus:menus(id, name, dishes(*))
    &menus_f.name=like.Comida%20Cubana
```

**Comment by steve-chavez** [comment, marked as answer]
> Is it possible to do this join without selecting it?

Yes, you can leave the parentheses empty for that. See https://github.com/PostgREST/postgrest/pull/2574.

This is available on the latest pre-release https://github.com/PostgREST/postgrest/releases/tag/v10.1.1.20221215

**Reply by lazaroofarrill** [reply]
> > Is it possible to do this join without selecting it?
> 
> 
> 
> Yes, you can leave the parentheses empty for that. See https://github.com/PostgREST/postgrest/pull/2574.
> 
> 
> 
> This is available on the latest pre-release https://github.com/PostgREST/postgrest/releases/tag/v10.1.1.20221215

I see, I'm using supabase so I guess I'll have to wait a while to get this working. Thank you.
